src/MaaCore/Task/SSS/SSSBattleProcessTask.cpp

```diff
@@ -199,16 +199,55 @@ bool asst::SSSBattleProcessTask::check_and_do_strategy(const cv::Mat& reusable)
     }
 
     for (auto& strategy : m_sss_combat_data.strategies) {
+        // 步骤(strategy)锁，同格子强制顺序执行
+        auto it = m_sss_combat_data.order.find(strategy.location);
+        if (*(it->second.front()) != strategy) {
+            continue;
+        }
+
         bool use_the_core = ranges::all_of(strategy.tool_men, [](const auto& pair) { return pair.second <= 0; }) &&
                             !strategy.core.empty() && exist_core.contains(strategy.core);
         if (use_the_core) {
             const auto& core = exist_core.at(strategy.core);
+            // 全局保留core以备暴毙、多位置放置相同core等情况
+            if (strategy.core_deployed && core.role != Role::Drone && core.role != Role::Unknown) {
+                strategy.core_deployed = false;
+                for (auto& strategy_reset : m_sss_combat_data.strategies) {
+                    if (strategy.location != strategy_reset.location) {
+                        continue;
+                    }
+                    for (auto& same_location_strategy : (it->second)) {
+                        if ((*same_location_strategy).index != strategy_reset.index) {
+                            continue;
+                        }
+                        bool skip = false;
+                        for (auto& [role, quantity] : (*same_location_strategy).tool_men) {
+                            if (role == Role::Drone || role == Role::Unknown) {
+                                skip = true;
+                            }
+                        }
+                        if (skip) {
+                            continue;
+                        }
+                        strategy_reset = *same_location_strategy;
+                        break;
+                    }
+                }
+                (it->second).emplace_back((it->second).front());
+                (it->second).erase((it->second).begin());
+                return false;
+            }
+
             if (!core.available) {
                 // 直接返回，等费用，等下次循环处理部署逻辑
                 break;
             }
-            if (auto it = m_all_cores.find(strategy.core); it != m_all_cores.end()) {
-                m_all_cores.erase(it);
+            strategy.core_deployed = true;
+            if (it->second.size() >= 2) {
+                if ((*(it->second.front())).index < (**(std::next(it->second.begin(), 1))).index) {
+                    (it->second).emplace_back((it->second).front());
+                    (it->second).erase((it->second).begin());
+                }
             }
```

Comment:
```suggestion
            if (strategy_order.size() >= 2 && strategy_order[0]->index < strategy_order[1]->index) {
                strategy_order.emplace_back(strategy_order.front());
                strategy_order.erase(strategy_order.begin());
            }
```
搞不懂整那么复杂干什么，想必是gpt写的吧

Reply:
这掐头掐尾的，不整个queue?

Reply:
我觉得既然是实现循环队列（是吗？），不如直接设一个变量表示当前位置